### ISSUE TITLE
docker 1.10-rc1 (and master) can't pull from quay.io

### ISSUE THREAD

**Issue by vikstrous** [docker/docker#19526]
```
docker pull quay.io/coreos/etcd:v2.2.4
Error response from daemon: unsupported manifest mediatype: text/html
```
```
docker version
Client:
 Version:      1.10.0-dev
 API version:  1.23
 Go version:   go1.5.3
 Git commit:   ee001d8
 Built:        Thu Jan 21 01:02:46 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-dev
 API version:  1.23
 Go version:   go1.5.3
 Git commit:   ee001d8
 Built:        Thu Jan 21 01:02:46 2016
 OS/Arch:      linux/amd64
```

**Comment by thaJeztah** [comment]
Looks like they're sending incorrect content-type headers?

**Comment by thaJeztah** [comment]
@aaronlehmann I assigned you for now, but let me know if you're not able to look into this

**Comment by HackToday** [comment]
Seems add 
err = distribution.RegisterManifestSchema("text/html", schema1Func)

can make it work. :)

**Comment by thaJeztah** [comment]
@HackToday it probably will, however, we should really think if that's a desired fix. The Content-Type header is used to determine the type of manifest that's sent, e.g. https://github.com/docker/docker/blob/7657e233f7dae10d8bacbd76460a1e4ca84a7ac7/vendor/src/github.com/docker/distribution/manifest/schema1/manifest.go#L13-L21

Loosening up validation can potentially have undesirable side-effects. Take, for example, a situation where a proxy server is used, and returns a HTML login-page; we don't want docker accept that blindly, and trying to download the HTML.

Basically it looks like quai.io is not a compliant registry, and doesn't implement the specifications correctly. This *should* be fixed in quai.io, not in Docker.

**Comment by aaronlehmann** [comment]
I think that given the problems we're finding with third-party registry implementations, it's safest to assume the payload is in the schema1 format (pre-Docker-1.10) if we don't recognize the content type. Any other ideas?

**Comment by stevvooe** [comment]
@aaronlehmann @thaJeztah We were intentionally vague about the initial manifest media type before supporting multiple versions. The current default in the PR is excellent at this stage, but we should eventually start maintaining a bound of supported versions.

The only problem I can see here is that, by default, the registry will return the earliest version (schema1). We may want to have the newer clients all declare `application/vnd.docker.*` (not sure if this is valid) in the `Accept` header to declare that they want the latest version of the manifest without specifying.

**Comment by aaronlehmann** [comment]
@stevvooe: I think it will be easier to get stricter about the content types as time passes. As third party registries gain support for schema2 they will have to handle media types correctly, and that will allow the engine to be stricter in the future.

**Comment by stevvooe** [comment]
@aaronlehmann Sounds good. That is the best approach for reliability.
